api/v1alpha1/tykoasapidefinition_types.go

```diff
@@ -95,6 +139,13 @@ type TykOasApiDefinition struct {
 	Status TykOasApiDefinitionStatus `json:"status,omitempty"`
 }
 
+func (t *TykOasApiDefinition) RemoveOASVersionStatus() {
```

Comment:
instead of setting each field to its empty values, i'd suggest setting the sub-struct to nil - so that in the future if we add/remove fields regarding versioning we won't forget to add a logic to delete them in this method.
For example, if all versioning-related information is stored in a struct, let's say `*VersioningStatus `, setting `t.Status.VersioningStatus = nil` is simpler